### DISCUSSION TITLE
OPTIONS *

### DISCUSSION THREAD

**Discussion by jschlyter** [aio-libs/aiohttp#5748] [Q&A]
I've been poking around with no success, trying here. Is there any way to request `OPTIONS *` using aiohttp?

**Comment by anesabml** [comment]
@jschlyter Could you please elaborate more about what are you trying to accomplish, if possible an example will be really helpful.

**Reply by jschlyter** [reply]
As part of implementing keep-alive for a long running process, we're sending periodic requests to keep the connection pool alive. We could send `OPTIONS /` or just any request, but sending the generic unspecified `OPTIONS *` just as web browsers would be a lot cleaner IMHO. It is indeed a very special snowflake, support would be useful.

**Reply by webknjaz** [reply]
Do you have an RFC link describing how this is supposed to work?

**Reply by jschlyter** [reply]
https://datatracker.ietf.org/doc/html/rfc7231#section-4.3.7

**Reply by webknjaz** [reply]
By the way, have you seen the API other web clients and servers implement? I'd be interested in seeing any prior art you can dig prior to deciding how to support this case. If there's something everyone else follows, we should try providing similar experience, if not, we'll need to think about the UX/DX aspect harder.

**Reply by jschlyter** [reply]
I tried to implement the same with `requests`, but failed. One way would be to follow `curl` and override the request target when sending the request, e.g. `curl --verbose --request OPTIONS --request-target "*" https://github.com/`.
